### ISSUE TITLE
Scroll position doesn't update when searching in Select2 4.0.0

### ISSUE THREAD

**Issue by cdaveb** [select2/select2#3465]
In Select2 4.0.0, when you are scrolled down near the bottom of a long select list and then you type in a search which changes the options, it does not refresh the scroll position to the top of the matched list. This can sometimes result in you ending up viewing different content than what you were viewing (because the list length is changing) but still ending up having to scroll upwards to see the beginning of the items matched by your search. 

I'm not sure if this is intentional behavior but it was confusing to one of our users as we were using Chosen before which did refresh the scroll position. I'm not sure if there's value in maintaining the old scroll position when the user may not even be viewing the same items they were when they started at that position. If this is intentional, perhaps it would be useful to have an option to change it to reset the scroll position to top when the search updates? I looked but I didn't see anything that looked like an option to do so.

I tested this behavior in Chrome and Firefox Mac and it performed the same in both.

Here's an example from the Select2 website:
1) Scroll down to the middle of the state list
![before](https://cloud.githubusercontent.com/assets/6133808/8095067/16d26568-0f80-11e5-98ee-24ccd224d36d.png)

2) Type the letter a in the search box
![after](https://cloud.githubusercontent.com/assets/6133808/8095077/2229fb4c-0f80-11e5-978c-9c2083ac1a8a.png)

You'll see you're in scrolled to a spot in the search results list that's completely different from where you started, but is also not at the top of the list, so it seems almost random.

**Comment by kevin-brown** [comment]
There was a reason for doing this, hidden deep within a commit message or ticket. I can't remember it though, so I'll let this stay open to see if we can work around the possible issues it would present.

**Comment by andrewpgordon** [comment]
I think that previous versions scrolled to top, and chosen uses this functionality as well.

**Comment by kelvin22** [comment]
Does anyone have a workaround for this?

**Comment by jasoncodes** [comment]
Here’s a workaround that I am using:

```coffee
$(el).data('select2').$dropdown.on 'input keyup', '.select2-search__field', ->
  $(this).closest('.select2-container').find('.select2-results > ul').scrollTop(0)
```

**Comment by andrewpgordon** [comment]
Thank you @jasoncodes for sharing that. Works like a charm.

**Comment by jackson-sandland** [comment]
@jasoncodes Could you clarify the notation you used above?

**Comment by jasoncodes** [comment]
@jackson-sand land It is [CoffeeScript](http://coffeescript.org/) which can be converted to JavaScript using their compiler.

**Comment by jasoncodes** [comment]
This issue appears to be fixed in 4.0.3.

**Comment by alexweissman** [comment]
Agreed, I cannot replicate this issue in 4.0.3.
